web/modules/custom/contact_form/src/Plugin/Menu/ContactSupportMenuLink.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\contact_form\Plugin\Menu;
+
+use Drupal\joinup_core\Plugin\Menu\DestinationAwareMenuLinkBase;
+
+/**
+ * Provides a custom menu link that leads to the Contact Joinup Support form.
+ */
+class ContactSupportMenuLink extends DestinationAwareMenuLinkBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getOptions(): array {
+    $options = parent::getOptions();
+    $options['query'] = $this->getDestinationArray() + $options['query'];
+    return $options;
+  }
```

Comment:
I think my proposal was not so clear. We should move this code in `DestinationAwareMenuLinkBase::getOptions()`. I think this is doing the whole job for the 2 classes (even EU Login needs that override). I didn't test this but, reading the code, it seems that `RedirectDestinationTrait` is already doing the job.